src/main/components/RequestBar.jsx

```diff
@@ -39,14 +40,39 @@ const RequestBar = (props) => {
     e.preventDefault();
   };
 
+  // Extra fetches for performance metrics
+  const fetchTimesList = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+  function getPerformanceMetricsData(getOrPost, sendingObj) {
+    let successfulFetchesCounter = 0;
+
+    function recordFetchTimes(i) {
+      fetch(uri, sendingObj)
+        .then(() => {
+          fetchTimesList[i] = new Date() - now;
+          successfulFetchesCounter += 1;
+          now = new Date();
+          if (successfulFetchesCounter === 9) {
```

Comment:
What will happen if a call is not successful ? 
Are you using the if statement so State is only set one time ?

Reply:
Yes, it's a good point-- if it'll sit until electron browser timeout.  worth taking another look at, though i think it's the desired behavior for now.  And yes, the if statement so it only gets updated once-- thinking about adding a spinning wheel until then.